pysmt/smtlib/script.py

```diff
@@ -392,3 +419,36 @@ def evaluate_command(cmd, solver):
                                   "Please open a bug-report." % cmd.name)
     else:
         raise UnknownSmtLibCommandError(cmd.name)
+
+def omt_evaluate_command(cmd, optimizer, client_data = None):
+
+    if cmd.name == smtcmd.MAXIMIZE:
+        rt = MaximizationGoal(cmd.args[0])
+        if client_data is not None:
+            client_data[0].append(rt)
+        return rt
+    elif cmd.name == smtcmd.MINIMIZE:
+        rt = MinimizationGoal(cmd.args[0])
+        if client_data is not None:
+            client_data[0].append(rt)
+        return rt
+    elif cmd.name == smtcmd.CHECK_SAT:
+        if client_data != None:
+            client_data[1].clear()
+            for g in client_data[0]:
+                client_data[1].append((g.term(), optimizer.optimize(g)[1]))
```

Comment:
Why? Can't you simply return the optimal model or a dictionary with the objectives values?

Reply:
it's necessary to return the model?